spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/security/BasicAuthSecurityConfiguration.java

```diff
@@ -157,6 +161,9 @@ protected void configure(HttpSecurity http) throws Exception {
 		http.addFilterBefore(sessionRepositoryFilter,
 				ChannelProcessingFilter.class).csrf().disable();
 		http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.IF_REQUIRED);
+
+		securityStateBean.setAuthenticationEnabled(true);
+		securityStateBean.setAuthorizationEnabled(true);
```

Comment:
I agree that the Spring Boot conditional annotations are not too easy on the brain.

Coming from the old Spring XD, we had originally used `security.basic.enabled` as the main driver to enable security in SCDF. That by itself is confusing also, also compounded by the fact that our OAuth2 configuration also uses `BasicAuth` which is completely unrelated to the normal Spring Security setup we use (File-based and Ldap authentication) that also uses Basic Auth.

As a auxiliary requirement (customer - @sabbyanandan), I removed the requirement to have `security.basic.enabled` to be present for OAuth2 configuration. In order to not break backwards-compatibility, you can still leave it there.

The key point is that `security.oauth2.client.client-id` is THE property that triggers OAuth2 configuration to be enabled. Also, either `BasicAuthSecurityConfiguration` is loaded OR `OAuthSecurityConfiguration` but never both.